components/constellation/constellation.rs

```diff
@@ -3270,7 +3265,8 @@ where
         browsing_context_id: BrowsingContextId
     ) -> bool {
         self.focused_browsing_context_id.map_or(false, |focus_ctx_id| {
-            self.fully_active_descendant_browsing_contexts_iter(browsing_context_id)
+            focus_ctx_id == browsing_context_id || self
```

Comment:
I ended up adding the optimization here rather than to the call site as it was too awkward that way with `focused_browsing_context_id` being an `Option`.